### DISCUSSION TITLE
creating an account requires clicking on link in invitation email

### DISCUSSION THREAD

**Discussion by tessus** [dani-garcia/vaultwarden#3320] [Q&A] [answered]
I am using vw `1.27.0-f2108990` and web-vault `2023.2.0`.

I have set the following in my config:

```
SIGNUPS_ALLOWED=false
SIGNUPS_VERIFY=false
SIGNUPS_DOMAINS_WHITELIST=example.com     # doesn't matter if this is set or not
SMTP_HOST=configured and able to send email
```

Now as an admin I invite a user with the email `user@not-example.com`.

I can only create an account, if I click on the link in the email. It is not possible to create an account for `user@not-example.com` by going to the web vault login and clicking on `Create account`. (Not setting `SIGNUPS_DOMAINS_WHITELIST` makes no difference.)

The invitation is in the database, so the server should know that the user is allowed to create an account, yet the creation fails with: `An error has occurred. Registration not allowed or user already exists.`

Is this working as designed? IIRC according to previous discussions this should be possible.

**Comment by stefan0xC** [comment, marked as answer]
Since you have disabled signups you don't want just anybody to be able to register so this is working as intended. Signup is possible only for the invited person via the link (only if you had mail disabled it would work without the link).

**Reply by tessus** [reply]
> Since you have disabled signups you don't want just anybody to be able to register 

Of course not, but I invited that user via the admin panel. Thus this user should be in the database as invited/pending or whatever. There is no reason to use the link, since the user was legitimately invited. Any other user would not be able to register, because they were not invited (no webtoken or whatever in the database).

**Reply by stefan0xC** [reply]
If `user@not-example.com` got the link to use your server why wouldn't they use the link? I mean otherwise if you invite `user@not-example.com` there is nothing that prevents anyone else from registering that account and using it (which kinda defeats the purpose of setting signups_allowed to false, which is IMHO a major drawback of not having mails enabled).

Also using the link to register has the added benefit of automatically verifying the email address.

**Reply by tessus** [reply]
> there is nothing that prevents anyone else from registering that account and using it

This is true, however how would someone know that I have invited `user@not-example.com`? But I guess you are right. It makes sense.

Thanks.
